driver-mongodb/src/main/java/io/nosqlbench/driver/mongodb/MongoAction.java

```diff
@@ -0,0 +1,64 @@
+package io.nosqlbench.driver.mongodb;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.Timer;
+import com.mongodb.ReadPreference;
+import com.mongodb.client.MongoDatabase;
+import io.nosqlbench.engine.api.activityapi.core.SyncAction;
+import io.nosqlbench.engine.api.activityapi.planning.OpSequence;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+
+public class MongoAction implements SyncAction {
+
+    private final static Logger logger = LoggerFactory.getLogger(MongoAction.class);
+
+    private final MongoActivity activity;
+    private final int slot;
+
+    private OpSequence<ReadyMongoStatement> sequencer;
+
+    public MongoAction(MongoActivity activity, int slot) {
+        this.activity = activity;
+        this.slot = slot;
+    }
+
+    @Override
+    public void init() {
+        this.sequencer = activity.getOpSequencer();
+    }
+
+    @Override
+    public int runCycle(long cycleValue) {
+        ReadyMongoStatement rms;
+        Bson queryBson;
+        try (Timer.Context bindTime = activity.bindTimer.time()) {
+            rms = sequencer.get(cycleValue);
+            queryBson = rms.bind(cycleValue);
+
+            if (activity.isShowQuery()) {
+                logger.info("Query(cycle={}):\n{}", cycleValue, queryBson);
```

Comment:
We might want to add a warning on this when it is enabled and cycles>1000 total, similar to how we do with the CQL activity.